sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/ProcessorLiveTests.cs

```diff
@@ -572,9 +572,15 @@ public async Task StopProcessingAllowsCallbackToComplete()
                 });
                 var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
 
+                int messageCount = 0;
                 async Task ProcessMessage(ProcessMessageEventArgs args)
                 {
-                    tcs.TrySetResult(true);
+                    var count = Interlocked.Increment(ref messageCount);
+                    if (count == 10)
```

Comment:
Should we have some kind of timeout safety here, so that the test doesn't hang if all messages aren't received?

Reply:
the test would timeout after 5 minutes.